src/Symfony/Component/Form/Extension/Core/Type/MultiStepType.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Extension\Core\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\Exception\InvalidOptionsException;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Silas Joisten <silasjoisten@proton.me>
+ * @author Patrick Reimers <preimers@pm.me>
+ * @author Jules Pietri <jules@heahprod.com>
+ */
+final class MultiStepType extends AbstractType
+{
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver
+            ->setRequired('steps')
+            ->setAllowedTypes('steps', 'array')
+            ->setAllowedValues('steps', static function (array $steps): bool {
+                foreach ($steps as $key => $step) {
+                    if (!\is_string($key)) {
+                        return false;
+                    }
+
+                    if ((!\is_string($step) || !is_subclass_of($step, AbstractType::class)) && !\is_callable($step)) {
```

Comment:
A `$step` object that implements `FormTypeInterface` should be allowed as well

Also, I have the feeling that parenthesis should be added here (mixing `||` and `&&`).